### PR TITLE
Buffer Unity SDK ReceiveData when watching for configuration changes

### PR DESCRIPTION
**Pull request by ZeroParticle**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:
Currently the Unity SDK watches for changes to the GameServer configuration by processing incoming data using `ReceiveData` on `DownloadHandlerScript`. This function is not guaranteed to be passed a single, complete message, so it is possible for the JSON parse to fail. Though extremely rare, I have run into this problem a few times while doing load tests.

**Which issue(s) this PR fixes**:

No associated issue, but I can open one if needed.

**Special notes for your reviewer**:
The Kubernetes JS client does something similar to this using the `byline` package: https://github.com/kubernetes-client/javascript/blob/master/src/watch.ts#L116

In my testing it appears that creating messages based on newline characters is enough, so I implemented only that case.

Additional documentation on the Unity ReceiveData function: https://docs.unity3d.com/2021.3/Documentation/ScriptReference/Networking.DownloadHandler.ReceiveData.html

### CODE REVIEW

**Comment by igooch** [comment]
Do you have someone on your team that can help out with reviewing this code? Our regular maintainers aren't as familiar with the Unity SDK.

**Review by liamlutton — APPROVED** [review]

File: sdks/unity/AgonesSdk.cs
Comment (igooch) [inline sdks/unity/AgonesSdk.cs:384]: What does the input look like for this method? Is all the data expected to be on one line?

**Review by igooch — COMMENTED** [review]
It would be nice to have at least a unit test for this. Although I'm not seeing any tests for the Unity SDK, which is... odd.

File: sdks/unity/AgonesSdk.cs
Comment (ZeroParticle) [reply sdks/unity/AgonesSdk.cs:384]: fullLine should always be a single JSON message on a single line. grpc-gateway turns streams into newline delineated JSON messages and JSON encoding will escape any newlines in the data, so the only newlines we encounter will be the ones intentionally sent to break full JSON messages.

**Review by igooch — APPROVED** [review]
Approving with the caveat that I don't have a way of testing this. LGTM per [Liam Lutton](https://github.com/liamlutton) and [Jack Schaffner](https://github.com/ZeroParticle).

**Pull request merged**
